docs-website/topics/serialization-customization-options.md

```diff
@@ -0,0 +1,524 @@
+even when the property has a default value:
+
+```kotlin
+// Imports declarations from the serialization and JSON handling libraries
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+//sampleStart
+@Serializable
+data class Project(val name: String, val language: String = "Kotlin")
+
+fun main() {
+    val data = Json.decodeFromString<Project>("""
+        {"name":"kotlinx.serialization","language":null}
+    """)
+    println(data)
+    // JsonDecodingException
+}
+//sampleEnd
+```
+{kotlin-runnable="true" validate="false"}
+
+> If you need to handle `null` values from third-party JSON, you can [coerce them to a default value](serialization-json-configuration.md#coerce-input-values).
+> 
+> You can also [omit explicit `null` values](serialization-json-configuration#omit-explicit-nulls) from the encoded JSON with the `explicitNulls` property.
+>
+{style="tip"}
+
+### Initializers in optional properties
+
+When the input contains an optional property, its initializer isn't called.
+For this reason, avoid using code with side effects in property initializers.
+
+Here's an example:
+
+```kotlin
+// Imports declarations from the serialization and JSON handling libraries
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+//sampleStart
+fun computeLanguage(): String {
+    println("Computing")
+    return "Kotlin"
+}
+
+@Serializable
+// Skips the initializer if language is in the input
+data class Project(val name: String, val language: String = computeLanguage())
+
+fun main() {
+    val data = Json.decodeFromString<Project>("""
+        {"name":"kotlinx.serialization","language":"Java"}
+    """)
+    println(data)
+    // Project(name=kotlinx.serialization, language=Java)
+}
+//sampleEnd
+```
+{kotlin-runnable="true"}
+
+In this example, since the `language` property is specified in the input, the `Computing` string isn't printed
+in the output.
+
+### Manage the serialization of default properties with `@EncodeDefault`
+<primary-label ref="experimental-general"/>
+
+By default, JSON serialization excludes properties that have default values.
+This reduces the size of the serialized data and avoids unnecessary visual clutter.
+
+Here's an example where the `language` property is excluded from the output because its value equals the default:
+
+```kotlin
+// Imports declarations from the serialization and JSON handling libraries
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+//sampleStart
+@Serializable
+data class Project(val name: String, val language: String = "Kotlin")
+
+fun main() {
+    val data = Project("kotlinx.serialization")
+    println(Json.encodeToString(data))
+    // {"name":"kotlinx.serialization"}
+}
+//sampleEnd
+```
+{kotlin-runnable="true"}
+
+> Learn more about configuring this JSON behavior in the [Encode default values](serialization-json-configuration.md#encode-default-values) section.
```

Comment:
I would rephrase this as "You can change JSON setting to affect all properties with default values, learn here how" to highlight the fact that you don't have to put `@EncodeDefault` everywhere if you don't want to